### PR TITLE
Ha lib integration

### PR DESCRIPTION
**Pull request by rohan-puri**

### CODE REVIEW

File: usr/tgtd.c [hunk 1]
```diff
@@ -519,6 +535,203 @@ static int parse_params(char *name, char *p)
...
+
+	json_t *tname = json_object_get(root, "TargetName");
+	if (!json_is_string(tname)) {
+		set_err_msg(resp, TGT_ERR_INVALID_TARGET_NAME,
+			"TargetName is not string");
+		return HA_CALLBACK_CONTINUE;
+	}
+
+	memset(cmd, 0, sizeof(cmd));
+	snprintf(cmd, sizeof(cmd), "tgtadm --lld iscsi --mode target --op new"
```
Comment (prasad-joshi) [inline usr/tgtd.c]: You might want to check return value of snprintf. If thing u wanted in string wrapped.

File: usr/tgtd.c [hunk 2]
```diff
@@ -519,6 +535,203 @@ static int parse_params(char *name, char *p)
...
+
+	json_t *vmdkid = json_object_get(root, "VmdkID");
+	if (!json_is_string(vmdkid)) {
+		set_err_msg(resp, TGT_ERR_INVALID_VMDKID,
+			"VmdkID is not string");
+		return HA_CALLBACK_CONTINUE;
+	}
+
+	memset(cmd, 0, sizeof(cmd));
+	snprintf(cmd, sizeof(cmd), "tgtadm --lld iscsi --mode logicalunit --op new"
```
Comment (prasad-joshi) [inline usr/tgtd.c]: Check return value of snprintf

File: usr/tgtd.c [hunk 3]
```diff
@@ -519,6 +535,203 @@ static int parse_params(char *name, char *p)
...
+
+	return HA_CALLBACK_CONTINUE;
+}
+
+int tgt_ha_start_cb(const _ha_request *reqp,
+	_ha_response *resp, void *userp)
+{
+	struct _ha_instance *hap = (struct _ha_instance *) userp;
+
+	pthread_create(&ha_hb_tid, NULL, &ha_heartbeat, (void *)hap);
```
Comment (prasad-joshi) [inline usr/tgtd.c]: Check return value. What if thread didn't start?

Comment (prasad-joshi) [reply usr/tgtd.c]: What happens when this can is called multiple times?

File: usr/tgtd.c [hunk 4]
```diff
@@ -519,6 +535,203 @@ static int parse_params(char *name, char *p)
...
+	return HA_CALLBACK_CONTINUE;
+}
+
+int tgt_ha_stop_cb(const _ha_request *reqp,
```
Comment (prasad-joshi) [inline usr/tgtd.c]: What happens when this cb is called multiple times

File: usr/tgtd.c (hunk 1 above)
Comment (rohan-puri) [reply usr/tgtd.c]: done.

File: usr/tgtd.c (hunk 2 above)
Comment (rohan-puri) [reply usr/tgtd.c]: done.

File: usr/tgtd.c (hunk 3 above)
Comment (rohan-puri) [reply usr/tgtd.c]: return value checked, if failed, we return error. handled multiple times occurence too.

**Pull request merged**